### PR TITLE
Catch only if we have pokeballs. 

### PR DESCRIPTION
**Pull request by joychugh**

### CODE REVIEW

**Comment by joychugh** [comment]
Will catch with pokeballs, great balls, ultraballs, masterballs in that order.

File: pgoapi/pgoapi.py [hunk 1]
```diff
@@ -250,18 +267,31 @@ def nearby_map_objects(self):
...
-                spin_modifier= 0.850,
-                hit_pokemon=True,
-                normalized_hit_position=1,
-                encounter_id=encounter_id,
-                spawn_point_guid=spawn_point_guid,
-                ).call()['responses']['CATCH_POKEMON']
-            if "status" in r:
-                return r
+        pokeball = 1
+        if self.inventory.poke_balls >= 1:
```
Comment (joychugh) [inline pgoapi/pgoapi.py]: here we did 4 attempts earlier, do we want to try the same?

**Comment by joychugh** [comment]
Will attempt to catch 4 (or less if do not have pokeballs anymore)

**Comment by j-e-k** [comment]
Awesome, thanks for all the contributions!

**Pull request merged**